### DISCUSSION TITLE
Not able to remove shadow from Chromium's menus

### DISCUSSION THREAD

**Discussion by fedxyz** [yshui/picom#1223] [Q&A]
Hi,

I have problems with shadow effects on Chromium's menu, I am trying to exclude it from shadow.

On Chromium using `xprop | grep -i class`

returns me

```
WM_CLASS(STRING) = "chromium", "Chromium"
```

in `picom.conf`

I setup, under `shadow-exclude` section I tried (one of them at time)

```
"class_g = 'Chromium'"
"class_g ~= '^.*(C|c)hromium.*$'",
"class_g ~= '^(C|c)hromium$'",
```

and the same using `"name"`.

No luck, I still have an horrible shadow effect on Chromium.

What am I doing wrong?

Thanks for the help.

**Comment by fedxyz** [comment]
Testing, the setting that work is:

```
   "name = ''",
```

that makes no sense...

**Comment by yshui** [comment]
Could be a bug. What version of picom are you using? Can you upload your trace log?

**Comment by fedxyz** [comment]
I am using picom v9 (9-1 from apt-show), I attach trace log file when the problem happen (I opened the popup menu in chromium), my config includes

```
   "class_g = 'Chromium'",
   "class_g ~= '^.*(C|c)hromium.*$'",
   "class_g ~= '^(C|c)hromium$'",
   "class_g *?= 'Chromium'",
   "class_g *?= 'Chrome'",
```
[problem.log](https://github.com/yshui/picom/files/14882576/problem.log)

**Comment by fedxyz** [comment]
Important thing that I didn't write before:  it worked fine without any rule for Chromium in the config but after a Chromium update I have this problem.

**Comment by sprig** [comment]
It has been a while, I don't know if still relevant for OP, but I ran into this thread while troubleshooting a similar issue, so if anyone else comes looking;
<img src="https://imgs.xkcd.com/comics/wisdom_of_the_ancients.png">

I used the `rules` config and unintuitively it turns out that the order matters and later, more general rules can override earlier, more specific rules. Make sure you disable shadow for chrome *after* and rules enabling shadow for general windows.

P.S.
For me specifically the class was a bit different - `chromium-browser-chromium`
